biogtr/training/train.py

```diff
@@ -0,0 +1,101 @@
+"""Training script for training and evaluating single GTR model."""
+from biogtr.config import Config
+from biogtr.datasets.tracking_dataset import TrackingDataset
+from biogtr.models.gtr_runner import GTRRunner
+from omegaconf import DictConfig
+import hydra
+import pytorch_lightning as pl
+import sys
+import torch
+import torch.multiprocessing
+
+device = "cuda" if torch.cuda.is_available() else "cpu"
+
+# useful for longer training runs, but not for single iteration debugging
+# finds optimal hardware algs which has upfront time increase for first
+# iteration, quicker subsequent iterations
+
+# torch.backends.cudnn.benchmark = True
+
+# pytorch 2 logic - we set our device once here so we don't have to keep setting
+torch.set_default_device(device)
+
+
+# not sure we need hydra? could just do argparse + omegaconf?
+@hydra.main(config_path="configs", config_name=None, version_base=None)
+def main(cfg: DictConfig):
+    """Main function for training.
+
+    Handles all config parsing and initialization then calls `trainer.train()`.
+
+    Args:
+        cfg: The config dict parsed by `hydra`
+    """
+    train_cfg = Config(cfg)
+
+    # update with extra cli args
+    hparams = {}
+    for arg in sys.argv[1:]:
+        if arg.startswith("+"):
```

Comment:
think we already discussed handling this better (e.g directly with hydra)?

Reply:
Yep see[here](https://github.com/talmolab/biogtr/blob/faa309bb012bc2e927f8db98ebba266ed07383cc/biogtr/training/train.py#L42)